compiler/AST/ForLoop.cpp

```diff
@@ -98,6 +119,21 @@ static void tryToReplaceWithDirectRangeIterator(Expr* iteratorExpr)
       Expr* high = range->get(2)->copy();
       iteratorExpr->replace(new CallExpr("chpl_direct_range_iter", low, high, stride));
     }
+
+    // replace a counted, low bounded range with unit stride with an equivalent
+    // direct range iter. e.g. replace:
+    //
+    //   `low..#count` (which is equivalent to `low..low+count-1`)
+    //
+    // with:
+    //
+    //   `chpl_direct_range_iter(low, low+count-1, 1)`
+    else if (count && range && range->isNamed("chpl_build_low_bounded_range"))
+    {
+      Expr* low = range->get(2)->copy();
+      Expr* high = new CallExpr("-", new CallExpr("+", low->copy(), count), new_IntSymbol(1));
+      iteratorExpr->replace(new CallExpr("chpl_direct_range_iter", low, high, stride));
```

Comment:
Is the reason to separate the definition of 'stride' (up above) from its single use here to support potentially supporting more complex expressions in the future?  Otherwise, it seems like hardcoding the "1" expression here in the new CallExpr() would be more straightforward given that this code isn't shared by any cases other than the lo..#count case (?).  I don't feel at all insistent about making this change, though, just editorializing.